addons/website/static/src/snippets/s_table_of_content/000.js

```diff
@@ -16,12 +16,20 @@ const TableOfContent = publicWidget.Widget.extend({
         this.$scrollingElement = this.$target.closest(".s_table_of_content").closestScrollable();
         this.previousPosition = -1;
         this._updateTableOfContentNavbarPosition();
-        extraMenuUpdateCallbacks.push(this._updateTableOfContentNavbarPosition.bind(this));
+
+        // bind _updateTableOfContentNavbarPosition.
+        this.boundUpdateNavbar = this._updateTableOfContentNavbarPosition.bind(this);
+        extraMenuUpdateCallbacks.push(this.boundUpdateNavbar);
     },
     /**
      * @override
      */
     destroy() {
+        // remove the bound function from the extraMenuUpdateCallbacks array so
+        // that `_updateTableOfContentNavbarPosition` won't gets called after
+        // public widget destroyed.
+        const indexOfCallback = extraMenuUpdateCallbacks.indexOf(this.boundUpdateNavbar);
+        extraMenuUpdateCallbacks.splice(indexOfCallback, 1);
         this.$target.css('top', '');
         this.$target.find('.s_table_of_content_navbar').css({top: '', maxHeight: ''});
         this._super(...arguments);
```

Comment:
Can you put that change in a separate commit. It's not about "remove ToC snippet on removing last content"
thanks :+1: